src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php

```diff
@@ -108,12 +108,38 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
     {
         $namespaceUseAnalyzer = new NamespaceUsesAnalyzer();
         $shortNames = [];
+        $currentSymbol = null;
+        $currentSymbolEndIndex = null;
 
         foreach ($namespaceUseAnalyzer->getDeclarationsFromTokens($tokens) as $namespaceUseAnalysis) {
             $shortNames[strtolower($namespaceUseAnalysis->getShortName())] = '\\'.strtolower($namespaceUseAnalysis->getFullName());
         }
 
+        $symbolKinds = [T_CLASS, T_INTERFACE];
+        if (\defined('T_ENUM')) { // @TODO drop the condition when requiring PHP 8.1+
+            $symbolKinds[] = T_ENUM;
+        }
+
         foreach ($tokens as $index => $token) {
+            if ($index === $currentSymbolEndIndex) {
+                $currentSymbol = null;
+                $currentSymbolEndIndex = null;
+
+                continue;
+            }
+
+            if ($token->isGivenKind(T_CLASS) && $tokens[$tokens->getPrevMeaningfulToken($index)]->isGivenKind(T_NEW)) {
```

Comment:
I think `TokensAnalyzer::isAnonymousClass` would be better here, it also deals with some attribute edge cases so you don have to do it here.
For example:
` new #[ExampleAttribute] class` is not supported here

Reply:
Nice spot! I added the test. It passed without change but this was accidental as the code used `{` as the symbol name for anonymous classes, which would never be used as a type so I still applied your suggestion :)

Reply:
ah right, yes, thanks! :) looking good, please move the test or tag it as PHP require 8.1 to make it pass, and than good to go from me!